common/src/main/java/com/wynntils/models/gear/SetModel.java

```diff
@@ -0,0 +1,108 @@
+import com.wynntils.core.net.UrlId;
+import com.wynntils.models.gear.type.GearSlot;
+import com.wynntils.models.gear.type.SetInfo;
+import com.wynntils.models.gear.type.SetInstance;
+import com.wynntils.models.stats.type.StatType;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class SetModel extends Model {
+    // Stored as a map for quick lookup <name, SetInfo>
+    private final Map<String, SetInfo> setData = new HashMap<>();
+    private final Map<GearSlot, SetInstance> setInstances = new EnumMap<>(GearSlot.class);
+
+    public SetModel() {
+        super(List.of());
+        loadSetData();
+    }
+
+    public SetInfo getSetInfoForId(String setId) {
+        return setData.getOrDefault(setId, null);
+    }
+
+    public SetInfo getSetInfoForItem(String itemName) {
+        return getSetInfoForId(getSetName(itemName));
+    }
+
+    /**
+     * @param itemName The name of the item to check
+     * @return The set name if the item is part of a set, null otherwise
+     */
+    public String getSetName(String itemName) {
+        for (Map.Entry<String, SetInfo> entry : setData.entrySet()) {
+            if (entry.getValue().items().contains(itemName)) {
+                return entry.getKey();
+            }
+        }
+        return null;
+    }
+
+    public void updateSetInstance(GearSlot slot, SetInstance instance) {
+        setInstances.put(slot, instance);
+    }
+
+    public void updateAllSetInstances(SetInstance instance) {
+        for (Map.Entry<GearSlot, SetInstance> entry : setInstances.entrySet()) {
+            if (entry.getValue()
+                    .getSetInfo()
+                    .name()
+                    .equals(instance.getSetInfo().name())) {
+                entry.getValue()
+                        .update(
+                                instance.getSetInfo(),
+                                instance.getActiveItems(),
+                                instance.getTrueCount(),
+                                instance.getWynnCount());
+            }
+        }
+    }
+
+    public Set<SetInstance> getUniqueSetInstances() {
+        return Set.copyOf(setInstances.values());
+    }
+
+    private void loadSetData() {
+        Download dl = Managers.Net.download(UrlId.DATA_STATIC_ITEM_SETS);
+        dl.handleReader(reader -> {
+            Map<String, RawSetInfo> rawSets =
+                    WynntilsMod.GSON.fromJson(reader, new TypeToken<Map<String, RawSetInfo>>() {}.getType());
+            rawSets.forEach((setName, rawSetInfo) -> {
+                List<Map<StatType, Integer>> bonuses = rawSetInfo.bonuses.stream()
+                        .map(bonusPair -> {
+                            Map<StatType, Integer> bonusMap = new HashMap<>();
+                            bonusPair.forEach((statName, statValue) -> {
+                                StatType statType = Models.Stat.fromApiName(statName);
```

Comment:
So it looks like this is a dependency for loading this URL. This should work serially, like `IngredientInfoRegistry#onDataLoaded` (so just call this from that event).

Also, implement `reloadData` and call this method there too.

Reply:
Sorry I am not too certain on what you mean by this. I kind of just copied what is going on in the IngredientInfoRegistry. Not sure where `reloadData` would go. My attempt is the "serially load gear/set infos" commit. It seems to work, but not sure if you want the syntax some other way.